src/lib.rs

```diff
@@ -1003,15 +1002,31 @@ impl<'a> PollItem<'a> {
 ///
 /// The result, if not `Err`, indicates the number of poll items that have
 /// events signaled.
-pub fn poll(items: &mut [PollItem], timeout: i64) -> Result<i32> {
+pub fn poll_some(items: &mut [PollItem], nitems: u32, timeout: i64) -> Result<i32> {
```

Comment:
Is it intended to have this used from outside? I don't see the advantage over `poll(&items[..nitems])`.

If it's going to be kept, the `nitems` parameter should at least be made a usize.